src/stepfunctions/steps/sagemaker.py

```diff
@@ -88,6 +95,12 @@ def __init__(self, state_id, estimator, job_name, data=None, hyperparameters=Non
 
             kwargs[Field.Resource.value] = get_service_integration_arn(SAGEMAKER_SERVICE_NAME,
                                                                        SageMakerApi.CreateTrainingJob)
+        # sagemaker.workflow.airflow.training_config does not accept ExecutionInput as input, only str, dict,
+        # TrainingInput or FileSystemInput. Transform placeholder into JSONpath to be processed and included in
+        # generated Sagemaker parameters. Placeholder for uri str only:
+        data_placeholder = isinstance(data, (ExecutionInput, StepInput))
```

Comment:
Prefix boolean variables with `is_` so we know it's a boolean. Additionally, we should be able to use the parent class `Placeholder` for the type check. Repeats for other places where we are checking for Placeholders.
```suggestion
        is_data_placeholder = isinstance(data, (Placeholder))
```

Reply:
Agreed!
Will make the changes in the next commits.